### ISSUE TITLE
Media Gallery component does not work at all

### ISSUE THREAD

**Issue by zessx** [htmlburger/carbon-fields#734]
## Version

  - Carbon Fields: 3.2.1
  - WordPress: 5.2.2
  - PHP: 7.2

## Behavior
The Media Gallery component does not work at all.

## Container definition
```php
Container::make( 'theme_options', __( 'Theme Options', $domain ) )
    ->add_fields(array(
        Field::make( 'media_gallery', 'crb_media_gallery', __( 'Media Gallery' ) )
            ->set_type( 'image' )
    ));
```

## Comments
We never figured out how to make the Media Gallery component work. This has been tested with multiple versions of CF (3+) and multiples versions of WP (5+)

Every attempt has be solved with the following JS errors, as soon as we add (or upload) an image via the component:

```js
index.js:195 Uncaught TypeError: Cannot read property 'url' of undefined
    at eval (index.js:195)
    at Array.map (<anonymous>)
    at eval (index.js:164)
    at MediaLibrary (index.js:24)
    at updateFunctionComponent (react-dom.development.js:14217)
    at beginWork (react-dom.development.js:15075)
    at performUnitOfWork (react-dom.development.js:17820)
    at workLoop (react-dom.development.js:17860)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)

STACK TRACE:
(anonymous) @ index.js:195
(anonymous) @ index.js:164
MediaLibrary @ index.js:24
updateFunctionComponent @ react-dom.development.js:14217
beginWork @ react-dom.development.js:15075
performUnitOfWork @ react-dom.development.js:17820
workLoop @ react-dom.development.js:17860
callCallback @ react-dom.development.js:149
invokeGuardedCallbackDev @ react-dom.development.js:199
invokeGuardedCallback @ react-dom.development.js:256
replayUnitOfWork @ react-dom.development.js:17107
renderRoot @ react-dom.development.js:17979
performWorkOnRoot @ react-dom.development.js:18837
performWork @ react-dom.development.js:18749
performSyncWork @ react-dom.development.js:18723
requestWork @ react-dom.development.js:18592
scheduleWork @ react-dom.development.js:18401
enqueueSetState @ react-dom.development.js:12313
Component.setState @ react.development.js:372
onStoreChange @ index.js:165
(anonymous) @ registry.js:59
globalListener @ registry.js:58
(anonymous) @ namespace-store.js:72
dispatch @ redux.js:229
(anonymous) @ promise-middleware.js:25
(anonymous) @ resolvers-cache-middleware.js:50
(anonymous) @ namespace-store.js:148
proxyDispatch @ index.js:79
container.<computed> @ index.es.js:248
container.<computed> @ index.es.js:248
container.<computed> @ index.es.js:248
(anonymous) @ index.js:85
(anonymous) @ index.js:89
s @ backbone.min.js?ver=1.2.3:1
r @ backbone.min.js?ver=1.2.3:1
m @ backbone.min.js?ver=1.2.3:1
k.trigger @ backbone.min.js?ver=1.2.3:1
_onModelEvent @ backbone.min.js?ver=1.2.3:1
s @ backbone.min.js?ver=1.2.3:1
r @ backbone.min.js?ver=1.2.3:1
m @ backbone.min.js?ver=1.2.3:1
k.trigger @ backbone.min.js?ver=1.2.3:1
clickSelect @ media-views.js?ver=5.2.1:5436
click @ media-views.js?ver=5.2.1:5579
dispatch @ jquery.js?ver=1.12.4-wp:3
r.handle @ jquery.js?ver=1.12.4-wp:3
Show 16 more frames
react-dom.development.js:15749 
```

```js
The above error occurred in the <MediaLibrary> component:
    in MediaLibrary (created by WithEffects)
    in WithEffects (created by MediaGalleryField)
    in Sortable (created by MediaGalleryField)
    in MediaGalleryField (created by WithEffects)
    in WithEffects (created by WithState(WithEffects))
    in WithState(WithEffects)
    in Unknown (created by WithFilters(WithState(WithEffects)))
    in WithFilters(WithState(WithEffects)) (created by WithEffects)
    in WithEffects (created by ComponentWithDispatch)
    in ComponentWithDispatch (created by Context.Consumer)
    in WithDispatch(WithEffects) (created by ComponentWithDispatch)
    in ComponentWithDispatch (created by Context.Consumer)
    in WithDispatch(WithDispatch(WithEffects)) (created by WithEffects)
    in WithEffects (created by ComponentWithSelect)
    in ComponentWithSelect (created by Context.Consumer)
    in WithSelect(WithEffects) (created by ComponentWithDispatch)
    in ComponentWithDispatch (created by Context.Consumer)
    in WithConditionalLogic(WithDispatch(WithDispatch(WithEffects))) (created by ComponentWithDispatch)
    in ComponentWithDispatch (created by Context.Consumer)
    in WithDispatch(WithConditionalLogic(WithDispatch(WithDispatch(WithEffects)))) (created by ComponentWithSelect)
    in ComponentWithSelect (created by Context.Consumer)
    in WithField(WithConditionalLogic(WithDispatch(WithDispatch(WithEffects)))) (created by WithFilters(WithFilters(WithState(WithEffects))))
    in WithFilters(WithFilters(WithState(WithEffects))) (created by Container)
    in div (created by Field)
    in div (created by Field)
    in Field
    in Unknown (created by WithFilters(Field))
    in WithFilters(Field) (created by ComponentWithSelect)
    in ComponentWithSelect (created by Context.Consumer)
    in WithSelect(WithFilters(Field)) (created by WithFilters(WithSelect(WithFilters(Field))))
    in WithFilters(WithSelect(WithFilters(Field))) (created by ComponentWithDispatch)
    in ComponentWithDispatch (created by Context.Consumer)
    in WithDispatch(WithFilters(WithSelect(WithFilters(Field)))) (created by ComponentWithSelect)
    in ComponentWithSelect (created by Context.Consumer)
    in WithField(WithFilters(WithSelect(WithFilters(Field)))) (created by Container)
    in div (created by Container)
    in div (created by Container)
    in Container (created by WithEffects)
    in WithEffects (created by WithFilters(Container))
    in WithFilters(Container) (created by ComponentWithSelect)
    in ComponentWithSelect (created by Context.Consumer)
    in WithContainer(WithFilters(Container))

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.

STACK TRACE:
logCapturedError @ react-dom.development.js:15749
logError @ react-dom.development.js:15783
update.callback @ react-dom.development.js:16649
callCallback @ react-dom.development.js:11164
commitUpdateEffects @ react-dom.development.js:11203
commitUpdateQueue @ react-dom.development.js:11194
commitLifeCycles @ react-dom.development.js:16010
commitAllLifeCycles @ react-dom.development.js:17262
callCallback @ react-dom.development.js:149
invokeGuardedCallbackDev @ react-dom.development.js:199
invokeGuardedCallback @ react-dom.development.js:256
commitRoot @ react-dom.development.js:17458
completeRoot @ react-dom.development.js:18912
performWorkOnRoot @ react-dom.development.js:18841
performWork @ react-dom.development.js:18749
performSyncWork @ react-dom.development.js:18723
requestWork @ react-dom.development.js:18592
scheduleWork @ react-dom.development.js:18401
enqueueSetState @ react-dom.development.js:12313
Component.setState @ react.development.js:372
onStoreChange @ index.js:165
(anonymous) @ registry.js:59
globalListener @ registry.js:58
(anonymous) @ namespace-store.js:72
dispatch @ redux.js:229
(anonymous) @ promise-middleware.js:25
(anonymous) @ resolvers-cache-middleware.js:50
(anonymous) @ namespace-store.js:148
proxyDispatch @ index.js:79
container.<computed> @ index.es.js:248
container.<computed> @ index.es.js:248
container.<computed> @ index.es.js:248
(anonymous) @ index.js:85
(anonymous) @ index.js:89
s @ backbone.min.js?ver=1.2.3:1
r @ backbone.min.js?ver=1.2.3:1
m @ backbone.min.js?ver=1.2.3:1
k.trigger @ backbone.min.js?ver=1.2.3:1
_onModelEvent @ backbone.min.js?ver=1.2.3:1
s @ backbone.min.js?ver=1.2.3:1
r @ backbone.min.js?ver=1.2.3:1
m @ backbone.min.js?ver=1.2.3:1
k.trigger @ backbone.min.js?ver=1.2.3:1
clickSelect @ media-views.js?ver=5.2.1:5436
click @ media-views.js?ver=5.2.1:5579
dispatch @ jquery.js?ver=1.12.4-wp:3
r.handle @ jquery.js?ver=1.12.4-wp:3
Show 17 more frames
react-dom.development.js:18807 
```

```js
Uncaught TypeError: Cannot read property 'url' of undefined
    at eval (index.js:195)
    at Array.map (<anonymous>)
    at eval (index.js:164)
    at MediaLibrary (index.js:24)
    at updateFunctionComponent (react-dom.development.js:14217)
    at beginWork (react-dom.development.js:15075)
    at performUnitOfWork (react-dom.development.js:17820)
    at workLoop (react-dom.development.js:17860)
    at renderRoot (react-dom.development.js:17946)
    at performWorkOnRoot (react-dom.development.js:18837)

STACK TRACE:
(anonymous) | @ | index.js:195
(anonymous) | @ | index.js:164
MediaLibrary | @ | index.js:24
updateFunctionComponent | @ | react-dom.development.js:14217
beginWork | @ | react-dom.development.js:15075
performUnitOfWork | @ | react-dom.development.js:17820
workLoop | @ | react-dom.development.js:17860
renderRoot | @ | react-dom.development.js:17946
performWorkOnRoot | @ | react-dom.development.js:18837
performWork | @ | react-dom.development.js:18749
performSyncWork | @ | react-dom.development.js:18723
requestWork | @ | react-dom.development.js:18592
scheduleWork | @ | react-dom.development.js:18401
enqueueSetState | @ | react-dom.development.js:12313
Component.setState | @ | react.development.js:372
container.<computed> | @ | index.es.js:248
(anonymous) | @ | index.js:88
(anonymous) | @ | index.js:89
s | @ | backbone.min.js?ver=1.2.3:1
r | @ | backbone.min.js?ver=1.2.3:1
m | @ | backbone.min.js?ver=1.2.3:1
k.trigger | @ | backbone.min.js?ver=1.2.3:1
_onModelEvent | @ | backbone.min.js?ver=1.2.3:1
s | @ | backbone.min.js?ver=1.2.3:1
r | @ | backbone.min.js?ver=1.2.3:1
m | @ | backbone.min.js?ver=1.2.3:1
k.trigger | @ | backbone.min.js?ver=1.2.3:1
clickSelect | @ | media-views.js?ver=5.2.2:5436
click | @ | media-views.js?ver=5.2.2:5579
dispatch | @ | jquery.js?ver=1.12.4-wp:3
r.handle | @ | jquery.js?ver=1.12.4-wp:3
```

**Comment by zessx** [comment]
Additional data: the component only fails with images without thumbnails.

**Comment by zessx** [comment]
Fixed in 3.1.3
